news/6197.bugfix

```diff
@@ -0,0 +1,2 @@
+Fix the ``have_directory_for_build`` ``AssertionError`` when using
```

Comment:
Maybe rephrase this to:

```
Fix a crash where PEP 517-based builds would fail due to not finalizing
a build directory internally, resulting in an ``AssertionError``.
```

Reply:
I adjusted the news entry along these lines and added some more test cases (by parametrizing the original test).